packages/ckeditor5-code-block/src/codeblockcommand.js

```diff
@@ -115,6 +115,11 @@ export default class CodeBlockCommand extends Command {
 			writer.rename( block, 'codeBlock' );
 			writer.setAttribute( 'language', language, block );
 			schema.removeDisallowedAttributes( [ block ], writer );
+
+			// The `codeBlock` element should contain only $text nodes as its children. See #9567.
+			Array.from( block.getChildren() )
+				.filter( child => !child.is( '$text' ) )
+				.forEach( writer.remove.bind( writer ) );
```

Comment:
This is removing also the soft breaks: 

![](https://user-images.githubusercontent.com/1232187/116775776-12dfae80-aa65-11eb-96fb-3e9e529b1ec8.gif)

We should use schema to check for allowed children:

```suggestion
			// The `codeBlock` element should contain only $text nodes as its children. See #9567.
			Array.from( block.getChildren() )
				.filter( child => !schema.checkChild( block, child ) )
				.forEach( child => writer.remove( child ) );
```

![2021-05-01 10 09 13](https://user-images.githubusercontent.com/1232187/116775823-74a01880-aa65-11eb-9e25-99c68164873a.gif)